packages/core/src/test/shared/utilities/functionUtils.test.ts

```diff
@@ -107,3 +107,53 @@ describe('debounce', function () {
         })
     })
 })
+
+describe('throttle', function () {
+    let counter: number
+    let fn: () => Promise<number>
+    let clock: ReturnType<typeof installFakeClock>
+
+    const callAndSleep = async (delayInMs: number) => {
+        const r = await fn()
+        await clock.tickAsync(delayInMs)
+        return r
+    }
+
+    const callAndSleepN = async (delayInMs: number, n: number) => {
+        const results = []
+        for (const _ of Array.from({ length: n })) {
+            results.push(await callAndSleep(delayInMs))
+        }
+        return results
+    }
+
+    beforeEach(function () {
+        clock = installFakeClock()
+        counter = 0
+        fn = throttle(() => ++counter, 10)
+    })
+
+    afterEach(function () {
+        clock.uninstall()
+    })
+
+    it('prevents a function from executing more than once in the `delay` window', async function () {
```

Comment:
I think we should add a test that asserts the first function call is executed and resolved before the first interval even completes. I.e if we tick 1ms while the delay is 3ms, we will get a result